lib/topological_inventory/ansible_tower/operations/worker.rb

```diff
@@ -4,29 +4,38 @@
 require "topological_inventory/ansible_tower/connection_manager"
 require "topological_inventory/ansible_tower/messaging_client"
 require "topological_inventory/providers/common/operations/health_check"
+require "topological_inventory/providers/common/operations/async_worker"
 
 module TopologicalInventory
   module AnsibleTower
     module Operations
       class Worker
         include Logging
 
+        ASYNC_MESSAGES = %w[Source.availability_check].freeze
+
         def run
-          TopologicalInventory::AnsibleTower::ConnectionManager.start_receptor_client
-          sleep 5
+          start_workers
           logger.info("Topological Inventory AnsibleTower Operations worker started...")
 
           client.subscribe_topic(queue_opts) do |message|
             log_with(message.payload&.fetch_path('request_context', 'x-rh-insights-request-id')) do
-              model, method = message.message.to_s.split(".")
-              logger.info("Received message #{model}##{method}, #{message.payload}")
-              process_message(message)
+              if ASYNC_MESSAGES.include?(message.message)
+                logger.debug("Queuing #{message.message} message for asynchronous processing...")
+                async_queue << message
+              else
+                model, method = message.message.to_s.split(".")
+                logger.info("Received message #{model}##{method}, #{message.payload}")
+
+                process_message(message)
+              end
             end
           end
         rescue => err
           logger.error("#{err.cause}\n#{err.backtrace.join("\n")}")
         ensure
           client&.close
+          async_worker&.exit
```

Comment:
`async_worker&.stop` will be better as this variable should be an instance of AsyncWorker